src/ol/source/wmtssource.js

```diff
@@ -73,6 +74,13 @@ ol.source.WMTS = function(options) {
 
   /**
    * @private
+   * @type {?number}
```

Comment:
As I've learned from @elemoine recently: Primitive types are not nullable by default, so I think that you can safely remove the exclamation mark.